addons/l10n_fr_pos_cert/static/src/js/l10n_fr_pos_cert.js

```diff
@@ -0,0 +1,244 @@
+                order.certification_text = '';
+            } else {
+                // We add a tag that will be replaced after, because
+                // when export_for_printing is called, hash is unknown
+                order.certification_text = "__CERTIFICATION_TEXT__";
+            }
+            return order;
+        },
+
+        export_as_JSON: function() {
+            var order = OrderParent.export_as_JSON.apply(this, arguments);
+            order.certification_text = this.get('certification_text');
+            return order;
+        },
+
+    });
+
+    /*************************************************************************
+        Extend module.PosModel:
+            - Overload load_server_data(). Load extra certification settings
+              at the end of the PoS load.
+            - Overload _save_to_server() and store if the order has been
+              correctly created in the promise 'certification_deferred'
+     */
+    var PosModelParent = models.PosModel.prototype;
+    models.PosModel = models.PosModel.extend({
+
+        load_server_data: function(){
+            var self = this;
+
+            // Call super
+            var loaded = PosModelParent.load_server_data.apply(this, arguments);
+
+            // Load Extra Settings
+            return $.when(loaded).then(function(){
+                self.chrome.loading_message(_t('Loading Certification Setting'));
+                var PosConfigModel = new DataModel('pos.config');
+                return PosConfigModel.call(
+                    'get_setting_l10n_fr_print_hash', [self.config.id], false
+                ).then(function (setting) {
+                    self.config.l10n_fr_print_hash = setting;
+                });
+            });
+        },
+
+        _save_to_server: function (orders, options) {
+            var self = this;
+
+            // Get PoS Config Settings
+            var setting = self.config.l10n_fr_print_hash;
+
+            if (setting === 'no_print'){
+                return PosModelParent._save_to_server.apply(this, arguments);
+            }
+            // Create a new promise that will resolved after the call to get the hash
+            certification_deferred = new $.Deferred();
+
+            var current_order = self.get('selectedOrder');
+            if (current_order){
+                // Init hash (and description that will be used, if server is unreachable)
+                current_order.set_hash(false, setting);
+            }
+
+            return PosModelParent._save_to_server.apply(this, arguments).then(function(server_ids) {
+                if (server_ids) {
+                    if (server_ids.length > 0){
+                        // Try to get hash of saved orders, if required
+                        var posOrderModel = new DataModel('pos.order');
+                        return posOrderModel.call(
+                            'get_certification_information', [server_ids], false
```

Comment:
_save_to_server and get_certification_information are @multi methods. I don't get why we fill the hash only for the current order, and not for the potential other orders